t/is-and-extract-ipv6-gua.pl

```diff
@@ -0,0 +1,472 @@
+use Test::More;
+use B qw(perlstring);
+
+SKIP: { eval { require Test::Warnings; } or skip($@, 1); }
+eval { require 'ddclient'; } or BAIL_OUT($@);
+
+my @valid_ipv6_not_global = (
+    # Unassigned address
+    "::",
+    "::0",
+    "::0:0",
+    "::0:0:0:0:0:0:0",
+    "0:0:0:0:0:0:0:0",
+    "0::0",
+    # link local address
+    "::1",
+    "::0:0:0:01",
+    "::0:0:0:001",
+    "::0:0:0:0001",
+    "0::1",
+    "0:0::1",
+    "0:0:0:0:0:0:0:1",
+    # Must start with a non-zero xxxx::/16
+    "::123.123.123.123",
+    "::2222:3333:4444:5555:6666:123.123.123.123",
+    "::3333:4444:5555:6666:123.123.123.123",
+    "::4444:5555:6666:123.123.123.123",
+    "::5555:6666:123.123.123.123",
+    "::6666:123.123.123.123",
+    "::ffff:192.168.1.1",
+    # Unique local address (ULA)
+    "FC11:2222:3333:4444:5555:6666:7777:8888",
+    "FC11::6666:7777:8888",
+    "FC11::",
+    "FC11:2222::123.123.123.123",
+    "FD11:2222:3333:4444:5555:6666::8888",
+    "FD11:2222:3333:4444:5555::8888",
+    "FD11:2222::8888",
+    "FD11::3333:4444:5555:6666:123.123.123.123",
+    "FD11::4444:5555:6666:123.123.123.123",
+    "FD11::5555:6666:123.123.123.123",
+    "FD11::6666:123.123.123.123",
+    "FD11::8888",
+    # Link local unicast addresses
+    "FE81:2222:3333:4444:5555::7777:8888",
+    "FE81:2222:3333:4444::7777:8888",
+    "FE81::8888",
+    "FE91:2222:3333::7777:8888",
+    "FEA1:2222::7777:8888",
+    "FEB1::7777:8888",
+    # Multicast addresses
+    "FF11::",
+    "FF11:2222::",
+    "FF11:2222:3333::",
+    "FF11:2222:3333:4444::",
+    "FF11:2222:3333:4444:5555::",
+    "FF11:2222:3333:4444:5555:6666::",
+    "FF11:2222:3333:4444:5555:6666:7777::",
+    "FF11:2222:3333:4444:5555:6666:7777:8888",
+);
+
+my @valid_ipv6_global = (
```

Comment:
Reduce this to addresses just outside the known-non-global blocks:

```perl
my @valid_ipv6_global = (
    "::2",
    "fbff:ffff:ffff:ffff:ffff:ffff:ffff:ffff",  # just before fc00::/7 ULA block
    "fe00::",                                   # just after  fc00::/7 ULA block
    "fe7f:ffff:ffff:ffff:ffff:ffff:ffff:ffff",  # just before fe80::/10 link-local block
    "fec0::",                                   # just after  fe80::/10 link-local block
    "feff:ffff:ffff:ffff:ffff:ffff:ffff:ffff",  # just before ff00::/8 multicast block
);
```

Reply:
Done, but I have kept some of the other valid IPv6 globals because I don't think just testing on those "outside the range" is valid test.  I don't think we will ever see globally routable addresses like these. Anything in the f000::/4 range will probably the the last assigned.  I did not include ::2 as it is not globally routable.